app/controllers/api/base_controller/renderer.rb

```diff
@@ -180,7 +180,24 @@ def expand_subcollections(json, type, resource)
       end
 
       def expand_subcollection?(sc, target)
-        respond_to?(target) && (@req.expand?(sc) || collection_config.show?(sc))
+        return false unless respond_to?(target) # If there's no query method, no need to go any further
+        expand_resources?(sc) || expand_action_resource?(sc) || resource_requested?(sc)
+      end
+
+      # Expand if: expand='resources' && no attributes specified && subcollection is configured
+      def expand_resources?(sc)
+        @req.expand?('resources') && @req.attributes.empty? && collection_config.show?(sc)
+      end
+
+      # Expand if: resource is being returned and subcollection is configured
+      # IE an update to /service_catalogs expects service_templates as part of its resource
+      def expand_action_resource?(sc)
+        @req.method != :get && collection_config.show?(sc)
```

Comment:
Without this it returns the subcollection - ie https://github.com/ManageIQ/manageiq/pull/14734/files#diff-d54e0c907bbc882d070223cd425c086bR90 fails because it returns `features` and https://github.com/ManageIQ/manageiq/pull/14734/files#diff-4dcc5ad5710cf94c07b7143879fef0a5R35 fails because it returns `service_templates`.